client/cmd/dexcctl/main.go

```diff
@@ -41,6 +43,62 @@ func main() {
 	}
 }
 
+// promptPasswords is a map of routes to password prompts. Passwords are
+// prompted in the order given.
+var promptPasswords = map[string][]string{
+	"openwallet": {"App password:"},
+	"newwallet":  {"App password:", "Wallet password:"},
+	"init":       {"Set new app password:"},
+	"register":   {"App password:"},
+}
+
+// swapCerts is a map of routes to tsl certificate indexes.
+var swapCerts = map[string]int{
+	"preregister": 1,
+	"register":    2,
+}
+
+// promptPWs prompts for passwords on stdin and returns an error if prompting
+// fails or a password is empty. Returns passwords as a slice of strings.
+func promptPWs(cmd string) ([]string, error) {
+	prompts, exists := promptPasswords[cmd]
+	if !exists {
+		return nil, nil
+	}
+	pws := make([]string, len(prompts))
+	// Prompt for passwords one at a time.
+	for i, prompt := range prompts {
+		pw, err := admin.PasswordPrompt(prompt)
+		if err != nil {
+			return nil, err
+		}
+		pws[i] = string(pw)
+		admin.ClearBytes(pw)
+	}
+	return pws, nil
+}
+
+// swapPathsForFiles will find tsl certificate paths for routes in swapCerts,
+// attempt to read those files, and swap the path for the entire certificate as
+// a string.
+func swapPathsForFiles(cmd string, args *[]string) error {
```

Comment:
Doesn't need to be a pointer to a slice.  You can modify the slice contents and it will reflect in the caller as long as you don't reallocate in the function.